extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -697,37 +684,48 @@ public String getErrorMessage() {
     private static final String HTTPS = "https:";
     private static final String DEOBFUSCATION_FUNC_NAME = "deobfuscate";
 
-    private final static String[] REGEXES = {
+    private static final String[] REGEXES = {
             "(?:\\b|[^a-zA-Z0-9$])([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)",
             "([\\w$]+)\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
             "\\b([\\w$]{2})\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
-            "yt\\.akamaized\\.net/\\)\\s*\\|\\|\\s*.*?\\s*c\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\(",
             "\\bc\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\("
     };
+    private static final String STS_REGEX = "signatureTimestamp[=:](\\d+)";
 
     @Override
     public void onFetchPage(@Nonnull final Downloader downloader)
             throws IOException, ExtractionException {
-        initialAjaxJson = getJsonResponse(getUrl() + "&pbj=1", getExtractorLocalization());
-
-        initialData = initialAjaxJson.getObject(3).getObject("response", null);
-        if (initialData == null) {
-            initialData = initialAjaxJson.getObject(2).getObject("response", null);
-            if (initialData == null) {
-                throw new ParsingException("Could not get initial data");
-            }
+        final String videoId = getId();
+        final Localization localization = getExtractorLocalization();
+        final ContentCountry contentCountry = getExtractorContentCountry();
+        final byte[] body = JsonWriter.string(prepareJsonBuilder(localization, contentCountry)
+                .value("videoId", videoId)
+                .done())
+                .getBytes(UTF_8);
+
+        // This boolean is needed if we don't want to fetch again the JSON player if the sts string
+        // is not null.
+        boolean stsKnown = false;
```

Comment:
Why save a boolean value here? (...see my next comment)

Reply:
I used what you said in https://github.com/TeamNewPipe/NewPipeExtractor/pull/604#discussion_r646097890, which is better.